library/core/src/ffi/c_str.rs

```diff
@@ -679,3 +679,35 @@ impl AsRef<CStr> for CStr {
         self
     }
 }
+
+/// Calculate the length of a nul-terminated string. Defers to C's `strlen` when
+/// possible,
+///
+/// # Safety
+///
+/// The pointer must point to a valid C string with a NUL terminator
```

Comment:
Should this method have the `isize::MAX` safety condition explicitly specified? It's implicit if "valid C string" actually means "a pointer that could be used to construct a CStr", but it might be worth documenting here (for `ptr::add`).

Reply:
I did not add it since this function can technically go up to `usize::MAX` so I don't think it's a constraint here - but I can clarify that "valid C string" actually just means "buffer with a NUL somewhere", i.e. not necessarily something that also meets CStr's requirements.

Reply:
`ptr::add` requires that the resulting pointer be in bounds of the same object as the start pointer, and we require that objects not be larger than `isize::MAX`.  I suppose the way this is accessed, the requirement is that the first NUL must be within the first `isize::MAX` bytes of the pointer.

I agree that "buffer with a NUL" (plus the constraint that the NUL be within `isize::MAX`) is more accurate as to the real requirements than "valid C string".